### PR TITLE
Re arrange flash layout to make linker happy

### PR DESCRIPTION
**Pull request by keith-packard**
Splitting data landing in flash resulted in the linker getting upset in a couple of ways.

First, the constructor/destructor arrays were no longer getting aligned via the ALIGN statement above them, resulting in the symbols being mis-aligned from the actual arrays.

Second, the constructors and destructors were somehow getting marked 'writable' which made the linker unhappy as that meant the contents of flash were marked both executable and writable.

Fix this up by stuffing the constructor/destructor arrays into the .text segment along with the ALIGN statement.

### CODE REVIEW

File: picolibc.ld.in [hunk 1]
```diff
@@ -113,6 +96,28 @@ SECTIONS
...
+		*(.srodata.cst8)
+		*(.srodata.cst4)
+		*(.srodata.cst2)
+		*(.srodata .srodata.*)
+
+	} >flash AT>flash :text
+
+	.data.rel.ro : {
+
+		/* relocation information (not generally used with picolibc) */
```
Comment (arichardson) [inline picolibc.ld.in:116]: Maybe rephrase this as "data that needs relocating" since the relocation information is in .rel(a).dun?

Comment (keith-packard) [reply picolibc.ld.in:116]: sure; I'm pretty unclear on what this section should contain, after all :-)

Comment (keith-packard) [reply picolibc.ld.in:116]: Are you otherwise happy with the change?

Comment (arichardson) [reply picolibc.ld.in]: Yes, everything else looks good to me.

Comment (arichardson) [reply picolibc.ld.in]: In general this section should be empty for most picolibc targets but it will be non-empty for PIC outputs or targets that require relocations for initializing some globals (e.g. CHERI enabled RISCV)
